<?xml version="1.0"?>
<launch>
  <!-- 设置仿真时间 -->
  <param name="/use_sim_time" value="true"/>

  <!-- 启动robot_state_publisher -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="50.0"/>
  </node>

  <!-- 启动map_server -->
  <node pkg="map_server" type="map_server" name="map_server" args="$(find robot_control_gui)/maps/test_map.yaml"/>

  <!-- 启动move_base -->
  <node pkg="move_base" type="move_base" name="move_base" output="screen">
    <rosparam file="$(find robot_control_gui)/config/costmap_common_params.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find robot_control_gui)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>
    <rosparam file="$(find robot_control_gui)/config/local_costmap_params.yaml" command="load"/>
    <rosparam file="$(find robot_control_gui)/config/global_costmap_params.yaml" command="load"/>
    <rosparam file="$(find robot_control_gui)/config/base_local_planner_params.yaml" command="load"/>
  </node>

  <!-- 启动测试节点 -->
  <test test-name="test_gui_functions" pkg="robot_control_gui" type="test_gui_functions" time-limit="60.0"/>
</launch> 